Microsoft.Toolkit/Diagnostics/ThrowHelper.ThrowExceptions.cs

```diff
@@ -38,33 +34,287 @@ private static string ToAssertString(this object? obj)
+        /// <exception cref="ObjectDisposedException">Thrown with <paramref name="objectName"/>.</exception>
+        [DoesNotReturn]
+        public static void ThrowObjectDisposedException(string objectName)
+        {
+            throw new ObjectDisposedException(objectName);
+        }
+
+        /// <summary>
+        /// Throws a new <see cref="ObjectDisposedException"/>.
+        /// </summary>
+        /// <param name="objectName">The name of the disposed object.</param>
+        /// <param name="message">The message to include in the exception.</param>
+        /// <exception cref="ObjectDisposedException">Thrown with <paramref name="objectName"/> and <paramref name="message"/>.</exception>
+        [DoesNotReturn]
+        public static void ThrowObjectDisposedException(string objectName, string message)
+        {
+            throw new ObjectDisposedException(objectName, message);
+        }
+
+        /// <summary>
+        /// Throws a new <see cref="OperationCanceledException"/>.
+        /// </summary>
+        /// <param name="message">The message to include in the exception.</param>
+        /// <exception cref="OperationCanceledException">Thrown with <paramref name="message"/>.</exception>
+        [DoesNotReturn]
+        public static void ThrowOperationCanceledException(string message)
+        {
+            throw new OperationCanceledException(message);
+        }
+
+        /// <summary>
+        /// Throws a new <see cref="OperationCanceledException"/>.
+        /// </summary>
+        /// <param name="token">The <see cref="CancellationToken"/> in use.</param>
+        /// <exception cref="OperationCanceledException">Thrown with <paramref name="token"/>.</exception>
+        [DoesNotReturn]
+        public static void ThrowOperationCanceledException(CancellationToken token)
+        {
+            throw new OperationCanceledException(token);
+        }
+
+        /// <summary>
+        /// Throws a new <see cref="ObjectDisposedException"/>.
+        /// </summary>
+        /// <param name="message">The message to include in the exception.</param>
+        /// <param name="token">The <see cref="CancellationToken"/> in use.</param>
+        /// <exception cref="ObjectDisposedException">Thrown with <paramref name="message"/> and <paramref name="token"/>.</exception>
+        [DoesNotReturn]
+        public static void ThrowThrowOperationCanceledException(string message, CancellationToken token)
+        {
+            throw new OperationCanceledException(message, token);
+        }
+
+        /// <summary>
+        /// Throws a new <see cref="PlatformNotSupportedException"/>.
+        /// </summary>
+        /// <param name="message">The message to include in the exception.</param>
+        /// <exception cref="PlatformNotSupportedException">Thrown with <paramref name="message"/>.</exception>
+        [DoesNotReturn]
+        public static void ThrowPlatformNotSupportedException(string message)
```

Comment:
What about `NotSupportedException`? I feel it is really essential alongside PNSE

Reply:
It's already there 😅

https://github.com/windows-toolkit/WindowsCommunityToolkit/blob/480b8609933e71cc363a0df58fbdcba1fb12d895/Microsoft.Toolkit/Diagnostics/ThrowHelper.ThrowExceptions.cs#L209

Reply:
Ahh, my bad 😄